lib/datasource/docker/index.ts

```diff
@@ -451,18 +451,46 @@ async function getTags(
     // AWS ECR limits the maximum number of results to 1000
     // See https://docs.aws.amazon.com/AmazonECR/latest/APIReference/API_DescribeRepositories.html#ECR-DescribeRepositories-request-maxResults
     const limit = ecrRegex.test(registry) ? 1000 : 10000;
-    let url = `${registry}/v2/${repository}/tags/list?n=${limit}`;
+
+    let url: string;
+    let page = 1;
+
+    // quay registry returns unordered tags using v2, let's use v1 for now
```

Comment:
Do not mix docker api v2 with quay rest api v1.

They are completely different and independent.

Reply:
@viceice any advice on how to use the quay rest api v1 anyhow? As described in https://github.com/renovatebot/renovate/issues/9176#issuecomment-800964821, the issue is that the v2 api is unsorted, in contrast to the v1 quay api.

Reply:
I'ts only your comment, which is wrong.

Currently renovate uses [docker registry api v2](https://docs.docker.com/registry/spec/api/). This doesn't define an order for the tags list. Most registries returning newest tags first, thats why if works for renovate, but quay.io returns oldest tags first and only allows 50 tags per page, so renovate can't find the newest tags.

So this feature uses the quay.io rest api v1 instead, which allows to get the tag list ordered, but this api can't be used for any other registry, as it's not the docker registry v1 api.

This is the cause why i prefer to movew the quay.io releated code to a new file `./quay.ts` and use that if the current passed registry is quay. this would also reduce the if-else statements. We maybe need to add other apis if find another registry, which is not behaving like the official docker registry.

Reply:
Got it. Moved to a separate function `getTagsQuayRegistry`. Feel free to further move/refactor as required!